src/cmd/flux-job.c

```diff
@@ -946,7 +1030,22 @@ void attach_exec_event_continuation (flux_future_t *f, void *arg)
     if (eventlog_entry_parse (o, &timestamp, &name, &context) < 0)
         log_err_exit ("eventlog_entry_parse");
 
-    if (!strcmp (name, "output-ready")) {
+    if (!strcmp (name, "input-ready")) {
+        if (json_unpack (context, "{s:i}",
+                         "leader-rank", &ctx->leader_rank) < 0)
+            log_err_exit ("error decoding input-ready context");
+
+        if (!(ctx->stdin_w = flux_fd_watcher_create (flux_get_reactor (ctx->h),
```

Comment:
Only concern is if user has stdin that has a line > buffer size.  But odds are perhaps low on that and perhaps this stdin line watcher is a net win.

Reply:
ended up there was an issue, the reactor incref/decref failed with a read buffer watcher.  The reason (I believe) is that the read buffer watcher has other watchers internal to it (prep, check, and I/O).  and I wasn't entirely sure how many there could be at any moment, b/c idle is sometimes turned on.

Perhaps active incref/decref should be a flag or something per watcher?  More complicated watchers would internally incref/decref as they start/stop watchers?

Reply:
> Perhaps active incref/decref should be a flag or something per watcher? More complicated watchers would internally incref/decref as they start/stop watchers?

This is what I had initially suggested, however @garlick had rightly pointed out that it would be too difficult to support "noref" watchers for the general case because of exactly this scenario. I haven't looked at the watcher implementation in awhile, but perhaps you'd have to expose a hook that would allow a watcher to get a callback if a request to make it not take *any* references on the reactor loop was made.

Hm, or maybe if there was a generic watcher flag, and a rule that a component watcher of any composite watcher *must* inherit this flag, *and* the flag can only be set at creation time, then that scheme could work out nicely (the ref/unref could be done in the low level watcher_start/stop code)

(I haven't looked at the code in awhile so the above could be a bad idea)

However, the ref/unref only working for simple watchers probably makes it useless except in the most trivial cases :-(. So maybe dropping the confusing manual refcount manipulation in favor of explicit ref/deref as is done in the shell is the right approach (maybe that could grow into a reactor and/or handle API to satisfy the common need?)

Reply:
For `flux-job` I think the ref/deref of the shell is a bit overkill.  All we want to do is stop watchers after both eventlogs are done.

For this purposes of this PR, I'm wondering if we should just revert 74f1e910e793e6d98840de7f10af3670d717c35d and go back to what was there before.  When both eventlog watchers are done, we had something setup to stop all of the watchers we cared to stop.  So adding the stdin watcher to the stop list is simple enough.

Reply:
That seems like it could work.